lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -58,13 +58,23 @@ def install!
                   generator.headers.concat module_map_additional_headers
                 end
                 create_umbrella_header(native_target) do |generator|
-                  generator.imports += if header_mappings_dir
-                                         file_accessors.flat_map(&:public_headers).map do |pathname|
-                                           pathname.relative_path_from(header_mappings_dir)
-                                         end
-                                       else
-                                         file_accessors.flat_map(&:public_headers).map(&:basename)
+                  generator.imports += file_accessors.flat_map do |file_accessor|
+                    header_dir = if !target.requires_frameworks? && dir = file_accessor.spec_consumer.header_dir
```

Comment:
I dont this this is correct -- can different subspecs really have a different `header_dir`?

Reply:
I guess so, as React Native has one.
https://github.com/facebook/react-native/blob/master/React.podspec#L244